firedrake/preconditioners/low_order.py

```diff
@@ -5,13 +5,17 @@
 
 
 class P1PC(PMGPC):
-    @staticmethod
-    def coarsen_element(ele):
+    def coarsen_element(self, ele):
         if isinstance(ele, MixedElement) and not isinstance(ele, (VectorElement, TensorElement)):
             raise NotImplementedError("Implement this method yourself")
 
-        p = ele.degree()
-        if p == 1:
+        N = ele.degree()
+        try:
+            N = min(N)
```

Comment:
Should we check here that they're all the same degree?
Like with `N, = set(N)` that you do below.

Reply:
Sure, this can be done here as well, since there is not built-in support for different degrees.